### PR TITLE
Add new constants for events

### PR DESCRIPTION
**Pull request by Sugirjan**
### Proposed changes in this pull request
-Add new constants for events names and event properties

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+public class AuthnDataPublisherProxy extends AbstractIdentityMessageHandler implements AuthenticationDataPublisher {
+    // Publish events and send them through IdentityEventService
+    private static final Log log = LogFactory.getLog(AuthnDataPublisherProxy.class);
+
+    @Override
+    public void publishAuthenticationStepSuccess(HttpServletRequest request, AuthenticationContext context,
+                                                 Map<String, Object> params) {
+        Event event = getEvent(request, context, null, params, IdentityEventConstants.Event
+                .AUTHENTICATION_STEP_SUCCESS);
```
Comment (Sugirjan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java]: rename the getEvent as createEvent

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        try {
+            FrameworkServiceDataHolder.getInstance().getIdentityEventService().handleEvent(event);
+        } catch (IdentityEventException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error is caught while handling the event: " + event.getEventName() + ".", e);
```
Comment (Sugirjan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java]: Add more details in error log

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                                                 Map<String, Object> params) {
+        Event event = getEvent(request, context, null, params, IdentityEventConstants.Event
+                .AUTHENTICATION_STEP_FAILURE);
+        try {
+            FrameworkServiceDataHolder.getInstance().getIdentityEventService().handleEvent(event);
+        } catch (IdentityEventException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error is caught while handling the event: " + event.getEventName() + ".", e);
+            }
+        }
```
Comment (Sugirjan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java]: Can put as a new private method

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    @Override
+    public boolean canHandle(MessageContext messageContext) {
+        return true;
+    }
+
+    private Event getEvent(HttpServletRequest request, AuthenticationContext context, SessionContext sessionContext,
+                           Map<String, Object> params, String eventName) {
+        //return event with required parameters as event properties.
```
Comment (Sugirjan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java]: put it as method comment

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        Map<String, Object> eventProperties = new HashMap<>();
+        eventProperties.put("request", request);
```
Comment (Sugirjan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AuthnDataPublisherProxy.java]: change as public constants

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java [hunk 6]
```diff
@@ -417,13 +417,12 @@ private void publishAuthenticationSuccess(HttpServletRequest request, Authentica
 
         AuthenticationDataPublisher authnDataPublisherProxy = FrameworkServiceDataHolder.getInstance()
                 .getAuthnDataPublisherProxy();
-        if (authnDataPublisherProxy != null && authnDataPublisherProxy.isEnabled(context)) {
+        if (authnDataPublisherProxy != null ) {
```
Comment (Sugirjan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java]: revert this line

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java [hunk 7]
```diff
@@ -82,6 +82,14 @@ private Event(){}
         public static final String POST_UPDATE_ROLE_LIST_OF_USER= "POST_UPDATE_ROLE_LIST_OF_USER";
         public static final String UPDATE_GOVERNANCE_CONFIGURATION= "UPDATE_GOVERNANCE_CONFIGURATION";
 
+        public static final String AUTHENTICATION_STEP_SUCCESS = "AUTHENTICATION_STEP_SUCCESS";
```
Comment (Sugirjan) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java]: it should be changed to enum.

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java [hunk 8]
```diff
@@ -86,6 +86,26 @@ private Event(){}
...
+
+        private final String eventName;
+
+        EventName(String eventName) {
+            this.eventName = eventName; }
+
+        @Override
+        public String toString() {
+            return eventName;
+        }
```
Comment (Sugirjan) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java]: redundant part

File: components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java [hunk 9]
```diff
@@ -86,6 +86,18 @@ private Event(){}
...
+    public enum EventName {
+
+        AUTHENTICATION_STEP_SUCCESS,
+        AUTHENTICATION_STEP_FAILURE,
+        AUTHENTICATION_SUCCESS,
+        AUTHENTICATION_FAILURE,
+        SESSION_CREATE,
+        SESSION_UPDATE,
+        SESSION_TERMINATE,
+
```
Comment (mefarazath) [inline components/identity-event/org.wso2.carbon.identity.event/src/main/java/org/wso2/carbon/identity/event/IdentityEventConstants.java]: Add a private constructor

**Pull request closed without merging**